src/drivers/uavcannode/UavcanNode.cpp

```diff
@@ -198,7 +205,12 @@ int UavcanNode::start(uavcan::NodeID node_id, uint32_t bitrate)
 	// Keep the bit rate for reboots on BenginFirmware updates
 	_instance->active_bitrate = bitrate;
 
-	_instance->ScheduleOnInterval(ScheduleIntervalMs * 1000);
+	if (node_id != 0) {
+		_instance->ScheduleOnInterval(ScheduleIntervalMs * 1000);
+
+	} else {
+		_instance->ScheduleNow();
```

Comment:
You could simply ScheduleNow() regardless then do the desired ScheduleOnInterval in !initialized in run.